src/node/rpc/member_frontend.h

```diff
@@ -973,16 +975,29 @@ namespace ccf
               &network,
               js_context);
             auto apply_func = js_context.function(
-              constitution,
-              "apply",
-              fmt::format("public:ccf.gov.constitution[0]", proposal_id));
+              constitution, "apply", "public:ccf.gov.constitution[0]");
 
+            JSValue argv[2];
             auto prop = JS_NewStringLen(
               js_context, (const char*)proposal.data(), proposal.size());
+            argv[0] = prop;
+
+            auto prop_id = JS_NewStringLen(
+              js_context, proposal_id.c_str(), proposal_id.size());
+            argv[1] = prop_id;
+
             auto val = js_context(
-              JS_Call(js_context, apply_func, JS_UNDEFINED, 1, &prop));
+              JS_Call(js_context, apply_func, JS_UNDEFINED, 2, argv));
+
+            // Execution of apply may make arbitrary changes to KV. Refetch
+            // current proposal info in case it has been modified
+            // NB: Can't do this, because we've only modified the .state on our
+            // local copy! pi_ = pi->get(proposal_id);
```

Comment:
Reminder to self: We should do this. Maybe not in this PR.

The use case we currently have is "`apply` may modify _every other_ proposal", but this is only a hair's breadth from "`apply` may modify _this_ proposal", and I think we should respect that modification here.

I'd like to audit anywhere else we may have this pattern of a long-held local copy of the KV value - we may need to `put` before `JS_Call`, and then `get` again afterwards.

Reply:
Tracked in #2454.